configure/lang/go/lang-go.Deployment.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    description: Go code intelligence provided by xlang-go, but supporting TLS and
+      WebSockets.
+  labels:
+    deploy: lang-go
+  name: lang-go
+spec:
+  minReadySeconds: 10
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: lang-go
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: lang-go
+    spec:
+      containers:
+      - args:
+        - go-langserver
+        - -mode=websocket
+        - -addr=:7777
+        - -usebuildserver
+        - -usebinarypkgcache=false
+        - -cachedir=$(CACHE_DIR)
+        - -freeosmemory=false
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: CACHE_DIR
+          value: /mnt/cache/$(POD_NAME)
+        image: sourcegraph/lang-go:insiders@sha256:8178238cb956676c990ad910a6c677293dca7e2660d9c72eeb6eed1836e62023
+        livenessProbe:
+          initialDelaySeconds: 5
+          tcpSocket:
+            port: lsp
+          timeoutSeconds: 5
+        name: lang-go
+        ports:
+        - containerPort: 7777
+          name: lsp
+        - containerPort: 6060
+          name: debug
+        readinessProbe:
+          tcpSocket:
+            port: 7777
+        terminationMessagePolicy: FallbackToLogsOnError
+        resources:
+          limits:
+            # TODO@ggilmore: Are these figures reasonable?
+            cpu: "4"
+            memory: 8Gi
+          requests:
+            cpu: "1"
+            memory: 2Gi
```

Comment:
These look reasonable to me compared to production https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/8f7a33406c6d2114abaa55e99a1b4cd849956dfe/configure/lang/go/lang-go.Deployment.yaml#L79-L84

Reply:
I am surprised we are setting these so low. I suspect any large repositories like Kubernetes will OOM with this few resources

Reply:
@slimsag  Do you think it's better to bump this to 12 gigs of memory, etc. even if not every user will need it?

Reply:
I think there is a high likelihood that users deploying a Kubernetes cluster AND Go language server will also have at least one very large repository. Basically I think the default is incorrectly inverted currently.

I would set to the values prod uses, then add a comment that you may lower them to the values currently here if you do not have large Go repositories.